### PR TITLE
GAMS Python API

### PR DESCRIPTION
**Pull request by aeseye**
GAMS Python API
  - To  Solve GAMS-based Optimization Problems from Python, on Eagle
  - Contains How To Guide and Demos.

### CODE REVIEW

File: languages/python/GAMS/demos/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+## 4. Set up Environment Parameter
+
+```
+export PYTHONPATH=$GAMS_PYTHON_API_FILES/api_37:$GAMS_PYTHON_API_FILES/gams
+```
+
+## 5. Load Solver
+
+Load the optimization solver you would like to use.
```
Comment (KevinSayers) [inline languages/python/GAMS/demos/README.md:47]: Link or list of solvers available or is Gurobi the only one installed on Eagle?

File: languages/python/GAMS/how_to_guide/gams_python_api_how_to_guide.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+- $GAMS_PYTHON_API_FILES/gams
+- $GAMS_PYTHON_API_FILES/api_[version-of-python]
+
+where version-of-python = 27, 36, 37, or 38 for python version 2.7, 3.6, 3.7, or 3.8, respectively. The python version can be obtained by using the command:
+
+```
+python --version
+```
+
+For example, if your python version is 3.7 the environment parameter $PYTHONPATH can be set using the following script:
```
Comment (KevinSayers) [inline languages/python/GAMS/how_to_guide/gams_python_api_how_to_guide.md:60]: I think it needs to be mentioned somewhere that you need to be part of whatever group to be able to access the GAMS python API files.

File: languages/python/GAMS/how_to_guide/gams_python_api_how_to_guide.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+Assuming you want to install your new environment in your scratch directory, you may want to create a directory that will contain all your Anaconda environments, e.g. /scratch/$USER/conda-envs/:
+
+```
+conda env create --prefix=/scratch/$USER/conda-envs/gams_python -f gams_python_api.yml
+```
+The installation may take a couple of minutes, wait until it is complete and you get a done message.
+
+# Load GAMS and Set up Environment Parameters
+
+Now, activate your Anaconda environment, load the appropriate GAMS version available on Eagle and establish the required environment varables. Follow the steps below.
```
Comment (KevinSayers) [inline languages/python/GAMS/how_to_guide/gams_python_api_how_to_guide.md:31]: I suggest including a bit more explanation. Something along the lines of "This Conda environment includes a basic Python environment. The GAMS API modules are then loaded directly from the module loaded on Eagle" 

Or somehow convey that the module load is actually where the API files are coming from and not a pip/conda install in the conda yaml file.

**Review by KevinSayers — CHANGES_REQUESTED** [review]
A couple quick change suggestions. Let me know if you want to discuss any of the points.

**Comment by KevinSayers** [comment]
Also perhaps rename the howto guide to README.md so it shows up without having to click the file in the repo.

File: languages/python/GAMS/demos/README.md (hunk 1 above)
Comment (aeseye) [reply languages/python/GAMS/demos/README.md:47]: Removed this part from this file and discussed about solvers in the how-to-guide read-me file.

File: languages/python/GAMS/how_to_guide/gams_python_api_how_to_guide.md (hunk 2 above)
Comment (aeseye) [reply languages/python/GAMS/how_to_guide/gams_python_api_how_to_guide.md:60]: Mentioned as a note in read-me file in the home directory of this repo.

File: languages/python/GAMS/how_to_guide/gams_python_api_how_to_guide.md (hunk 3 above)
Comment (aeseye) [reply languages/python/GAMS/how_to_guide/gams_python_api_how_to_guide.md:31]: Added this as a note in the same place.

File: languages/python/GAMS/demos/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+```
+module load gurobi/9.0.2
+```
```
[the remark this answers is not in the document]
Comment (aeseye) [reply languages/python/GAMS/demos/README.md:51]: Yes, no need to load solvers from Eagle modules. GAMS itself comes with almost all solvers. I updated this part. However, it is a must to load the GAMS module.

**Comment by aeseye** [comment]
> Also perhaps rename the howto guide to README.md so it shows up without having to click the file in the repo.

Renamed.

**Pull request merged**